cmd/agent/agent.go

```diff
@@ -47,7 +52,13 @@ func NewAgentCmd() *cobra.Command {
 
 	startCmd.Flags().StringVar(&sshAddress, "ssh-address", "", "The address to which the trento-agent should be reachable for ssh connection by the runner for check execution.")
 
-	startCmd.Flags().IntVarP(&discoveryPeriod, "discovery-period", "", 10, "Discovery mechanism loop period in seconds")
+	startCmd.Flags().IntVarP(&clusterDiscoveryPeriod, "cluster-discovery-period", "", 10, "Cluster discovery mechanism loop period in seconds")
```

Comment:
What hsppens if we get a non integer value in one of these options? For instance the string `whatever`.
And what happens if we get a `0` or `-1`?

:sweat_smile:

Reply:
In the first case viper/cobra should scream I hope. For the `0` or `-1` cases not sure :thinking:

Reply:
For -1: 
```
panic: non-positive interval for NewTicker

goroutine 6 [running]:
time.NewTicker(0xc0000bbca0)
	time/tick.go:24 +0x10f
github.com/trento-project/trento/internal.Repeat({0xef6be0, 0x14}, 0xc0004a5f00, 0xc0001880e0, {0x1a9b518, 0xc0000b8380})
	github.com/trento-project/trento/internal/utils.go:106 +0x7a
github.com/trento-project/trento/agent.(*Agent).startDiscoverTicker(0xc0001c2190, {0x1a92d90, 0xc0001c2230})
	github.com/trento-project/trento/agent/agent.go:108 +0x93
github.com/trento-project/trento/agent.(*Agent).Start.func1(0xc0000343d0, {0x1a92d90, 0xc0001c2230})
	github.com/trento-project/trento/agent/agent.go:75 +0xf1
created by github.com/trento-project/trento/agent.(*Agent).Start
	github.com/trento-project/trento/agent/agent.go:72 +0x209

```

We could add some validation so it doesn't reach the `NewTicker` call and we catch it in the cobra argument parsing. We need it anyway to implement the min values that @abravosuse suggested.

Reply:
Ok then, have a lot of fun :smile: